frontend/src/components/common/TagList/index.tsx

```diff
@@ -0,0 +1,40 @@
+import TagButton, { type TagButtonVariant } from '../TagButton';
+import * as S from './TagList.styled';
+import { HASHTAGS } from '@/constants/hashTags';
+
+interface TagListProps<T> {
+  tags: T[];
+  selectedTag: T;
+  setSelectedTag: (tag: T) => void;
+  variant?: TagButtonVariant;
+  keyName: keyof T;
+}
+
+export default function TagList<T extends { id: number }>({
+  tags,
+  selectedTag,
+  setSelectedTag,
+  variant,
+  keyName,
+}: TagListProps<T>) {
+  return (
+    <S.TagListContainer>
+      {tags.map((tag) => {
+        const name = tag[keyName] as string;
+        const isSelected = tag.id === selectedTag.id;
+        return (
+          <TagButton
+            key={tag.id}
+            isSelected={isSelected}
+            onClick={() =>
+              setSelectedTag(isSelected ? ({ id: 0, [keyName]: HASHTAGS.all } as T) : tag)
```

Comment:
```{ id: 0, [keyName]: HASHTAGS.all }``` <--요 객체가 어떤 의미인지 여기 코드만 보고는 이해가 잘 되지 않을 수 있을 것 같아요! 전체 선택하는 방법을 ```{ id: 0, [keyName]: HASHTAGS.all }```로 설정하는 방식이 아닌 다른 방식(ex. 비어있을 때는 전체 선택으로 간주)으로 바꿔도 좋을 것 같아요! 혹은 props로 setSelectedTag 대신 toggleSelectedTag와 같은 함수를 받아도 좋을 것 같아요 ㅎㅎ


1안
```suggestion
              setSelectedTag(isSelected ? null : tag)
```

2안
```suggestion
              toggleSelectedTag();
```